modules/core/react/lib/CanvasProvider.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+import {ThemeProvider} from 'emotion-theming';
+import InputProvider from './InputProvider';
+import {CanvasTheme} from './theming/types';
+import {defaultCanvasTheme} from './theming/theme';
+
+export interface CanvasProviderProps {
+  theme: CanvasTheme;
+}
+
+/**
+ * TODO:
+ * - Opt in for fonts
+ * - Make themes work without provider
+ */
+
+export default class CanvasProvider extends React.Component<CanvasProviderProps> {
```

Comment:
Not sure if we want this to be required or not. If we implicitly depend on `theme` in our components, consumers will get an error if they don't have a `CanvasProvider` HOC. Maybe we want to make this required anyway to enforce the `InputProvider`, but I added the `getTheme` function to work around [this](https://github.com/emotion-js/emotion/issues/1193) to keep it optional for now.